### PR TITLE
pin jsonschema to <4.18 in validation and test

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):**

- https://github.com/stac-utils/pystac/issues/1186

**Description:**

**PR Checklist:**

- [X] `pre-commit` hooks pass locally
- [X] Tests pass (run `scripts/test`)
- [X] Documentation has been updated to reflect changes, if applicable
- [X] This PR maintains or improves overall codebase code coverage.
- [X] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

**Review by jsignell — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+### Changed
+
+- Pin jsonschema version to <4.18 until regresssions are fixed
```
Comment (jsignell) [inline CHANGELOG.md:7]: Is #1189 not enough?

Comment (philvarner) [reply CHANGELOG.md:7]: If you installed `test` but not `validation`, you end up with a version that doesn't work.

Comment (philvarner) [reply CHANGELOG.md:7]: because the test restriction is `~=4.17` instead of `~=4.17.0`, so any 4.18.x is valid to install for that constraint

Comment (vincentsarago) [reply CHANGELOG.md:7]: IMO we should use the same bounds introduced in #1189

Comment (philvarner) [reply CHANGELOG.md:7]: that's probably better.

Comment (philvarner) [reply CHANGELOG.md:7]: done

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**